src/System.Reflection/tests/PropertyInfo/PropertyInfo_propertyTests.cs

```diff
@@ -13,230 +13,88 @@ namespace System.Reflection.Tests
 
             Assert.NotNull(pi);
-            Assert.False(pi.CanRead, "Failed!  CanRead Failed for write only property. Expected False , returned True");
+            Assert.Equal(boolean, pi.CanRead);
         }
 
-        //Verify CanWrite for read write PropertyInfo
-        [Fact]
-        public static void TestCanWrite1()
+        [Theory]
+        [InlineData("MyPropAA", true)]
+        [InlineData("MyPropBB", false)]
+        [InlineData("MyPropCC", true)]
+        public static void TestCanWrite(String propName, bool boolean)
         {
-            string propName = "MyPropAA";
             PropertyInfo pi = GetProperty(typeof(SampleProperty), propName);
 
-
             Assert.NotNull(pi);
-            Assert.True(pi.CanWrite, "Failed!  CanWrite Failed for read write property. Expected True , returned False");
-        }
-
-        //Verify CanWrite for readonly PropertyInfo
-        [Fact]
-        public static void TestCanWrite2()
-        {
-            string propName = "MyPropBB";
-            PropertyInfo pi = GetProperty(typeof(SampleProperty), propName);
+            Assert.Equal(boolean, pi.CanWrite);
 
-            Assert.NotNull(pi);
-            Assert.False(pi.CanWrite, "Failed!  CanWrite Failed for read only property. Expected False , returned True");
         }
 
-
-        //Verify CanWrite for writeonly PropertyInfo
-        [Fact]
-        public static void TestCanWrite3()
+        [Theory]
+        [InlineData("DerivedPropertyProperty", typeof(DerivedProperty), "DerivedProperty")]
+        [InlineData("BasePropertyProperty", typeof(BaseProperty), "BaseProperty")]
+        public static void TestDeclaringType(String propName, Type type, String typeName)
         {
-            string propName = "MyPropCC";
-            PropertyInfo pi = GetProperty(typeof(SampleProperty), propName);
-
-            Assert.NotNull(pi);
-            Assert.True(pi.CanWrite, "Failed!  CanWrite Failed for write only property. Expected True , returned False");
-        }
-
-        //Verify DeclaringType for PropertyInfo
-        [Fact]
-        public static void TestDeclaringType1()
-        {
-            string propName = "DerivedProeprtyProperty";
-            PropertyInfo pi = GetProperty(typeof(DerivedProeprty), propName);
-
-            Assert.NotNull(pi);
-            Assert.NotNull(pi.DeclaringType.Name);
-            Assert.Equal("DerivedProeprty", pi.DeclaringType.Name);
-        }
-
-        //Verify DeclaringType for PropertyInfo
-        [Fact]
-        public static void TestDeclaringType2()
-        {
-            string propName = "BasePropertyProperty";
-            PropertyInfo pi = GetProperty(typeof(BaseProperty), propName);
+            PropertyInfo pi = GetProperty(type, propName);
 
             Assert.NotNull(pi);
             Assert.NotNull(pi.DeclaringType.Name);
-            Assert.Equal("BaseProperty", pi.DeclaringType.Name);
+            Assert.Equal(typeName, pi.DeclaringType.Name);
```

Comment:
We probably don't need the ```Assert.NotNull(pi.DeclaringType.Name)``` assert... if it's null, it'll fail the subsequent Equal test anyway.